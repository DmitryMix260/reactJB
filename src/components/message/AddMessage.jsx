import { useState } from 'react';

export const AddMessage = ({ addMessage }) => {
  const [message, setMessage] = useState('');
  const handleAddMessage = (ev) => {
    ev.preventDefault();
    addMessage({
      autor: 'user',
      message,
    });
    setMessage('');
  };

  return (
    <>
      <form action="#" onSubmit={handleAddMessage}>
        <div>
          <p>Print message</p>
        </div>
        <input
          type="text"
          value={message}
          placeholder="Type message"
          onChange={(ev) => setMessage(ev.target.value)}
        />
        <button disabled={!message}>Submit</button>
      </form>
    </>
  );
};
