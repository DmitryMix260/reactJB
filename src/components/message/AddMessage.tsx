import React, { FC, useEffect, useState, useRef, memo } from 'react';
import { Button, TextField } from '@mui/material';
import { AUTHOR, Message } from 'src/types';

interface AddMessageProps {
  addMessage: (msg: Message) => void;
}

export const AddMessage: FC<AddMessageProps> = memo(({ addMessage }) => {
  const [message, setMessage] = useState('');
  const handleAddMessage = (ev: React.FormEvent<HTMLFormElement>) => {
    ev.preventDefault();
    addMessage({
      author: AUTHOR.USER,
      message,
    });
    setMessage('');
  };
  const inputFocus = useRef(null);

  useEffect(() => {
    inputFocus.current.focus();
  });

  return (
    <>
      <div>
        <p>Print message</p>
      </div>
      <form action="#" onSubmit={handleAddMessage} data-testid="addmessage">
        <TextField
          type="text"
          value={message}
          placeholder="Type message"
          onChange={(ev) => setMessage(ev.target.value)}
          role="input"
          ref={inputFocus}
          autoFocus={true}
        />
        <Button disabled={!message} role="button" type="submit">
          Submit
        </Button>
      </form>
    </>
  );
});
