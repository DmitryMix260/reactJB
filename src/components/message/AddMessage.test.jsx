import { render, screen } from '@testing-library/react';
import { AddMessage } from './AddMessage';
import '@testing-library/jest-dom';

describe(AddMessage, () => {
  it('render with snapshot', () => {
    const { asFragment } = render(<AddMessage disabled={false} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
