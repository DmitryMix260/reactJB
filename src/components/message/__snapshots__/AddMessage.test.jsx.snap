// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMessage render with snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div>
      <p>
        Print message
      </p>
    </div>
    <input
      placeholder="Type message"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
